components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/handlers/TokenResponseTypeHandler.java

```diff
@@ -137,7 +138,7 @@ public OAuth2AuthorizeRespDTO issue(OAuthAuthzReqMessageContext oauthAuthzMsgCtx
 
             AccessTokenDO existingAccessTokenDO = null;
             // check if valid access token exists in cache
-            if (cacheEnabled) {
+            if (cacheEnabled && isHashDisabled) {
```

Comment:
Checking for a access token from cache and databases need to be done if "isHashDisabled" is true. Shall we create a if block for isHashDisabled and add this if block and the next if block inside it?